src/agri_ai/core/optimized_database.py

```
@@ -0,0 +1,397 @@
+                use_cache=True
+            )
+            
+            logger.info(f"Retrieved {len(tasks)} tasks for worker {worker_id} on {date}")
+            return tasks
+            
+        except Exception as e:
+            logger.error(f"Error retrieving today's tasks: {e}")
+            raise DatabaseQueryError(
+                f"今日のタスク取得中にエラーが発生しました: {str(e)}",
+                context={"worker_id": worker_id, "date": date}
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)